compositor_render/src/scene/validation.rs

```diff
@@ -36,13 +36,16 @@ pub(super) fn validate_scene_update(
     old_outputs: &HashMap<OutputId, OutputScene>,
     updated_output: &OutputScene,
 ) -> Result<(), SceneError> {
-    let updated_outputs: Vec<&OutputScene> = old_outputs
-        .iter()
-        .map(|(id, output)| match id {
-            id if id == &updated_output.output_id => updated_output,
-            _ => output,
-        })
-        .collect();
+    let updated_outputs = match old_outputs.is_empty() {
+        true => vec![updated_output],
+        false => old_outputs
+            .iter()
+            .map(|(id, output)| match id {
+                id if id == &updated_output.output_id => updated_output,
+                _ => output,
+            })
+            .collect(),
+    };
```

Comment:
This won't work if there already are some outputs and a new one is added.

Reply:
Wojtek is correct, I addressed it in https://github.com/membraneframework/video_compositor/pull/375 so fill free to close it.